### ISSUE TITLE
Plugin doesn't checks type errors for const arguments in multi-segment paths

### ISSUE THREAD

**Issue by mchernyavsky** [intellij-rust/intellij-rust#8150]
## Environment

* **IntelliJ Rust plugin version:** 0.4.159.4191-212-nightly
* **Rust toolchain version:** 1.57.0-nightly (addb4da68 2021-09-25) x86_64-apple-darwin
* **IDE name and version:** CLion 2021.2.3 (CL-212.5457.51)
* **Operating system:** macOS 12.0.1
* **Macro expansion engine:** new
* **Name resolution engine:** new

## Problem description

The plugin doesn't report the type mismatch on line 9.

<img width="853" alt="Screenshot 2021-11-22 at 17 34 37" src="https://user-images.githubusercontent.com/6079006/142884572-34ed2406-c698-4299-89dd-7a052bfe2b7b.png">

## Steps to reproduce
```rust
struct S<const N: usize>;

impl<const N: usize> S<N> {
    fn foo() {}
    fn bar(&self) {}
}

fn main() {
    S::<"">::foo();
    S::<"">.bar();
}

```
